services/sky/document_view.h

```diff
@@ -41,14 +35,13 @@ class TextureLayer;
 
 class DocumentView : public blink::ServiceProvider,
                      public blink::SkyViewClient,
-                     public mojo::ViewManagerDelegate,
-                     public mojo::ViewObserver,
+                     public mojo::NativeViewportEventDispatcher,
                      public sky::LayerClient,
                      public sky::LayerHostClient,
                      public ui::GestureConsumer {
  public:
-  DocumentView(mojo::InterfaceRequest<mojo::ServiceProvider> services,
-               mojo::ServiceProviderPtr exported_services,
+  DocumentView(mojo::InterfaceRequest<mojo::ServiceProvider> exported_services,
+               mojo::ServiceProviderPtr imported_services,
```

Comment:
These are the serviceprovider pipes particular to a connection.  This connection will be from the shell (since it's a primordial app) but that doesn't mean that it's the thing to request services from the shell - the way to do that is to call Shell::ConnectToApplication().  So I think you should drop these.  If there's something I'm missing with this then please make the names match the mojom exactly.